beard/clustering/blocking.py

```diff
@@ -0,0 +1,204 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Beard.
+# Copyright (C) 2014 CERN.
+#
+# Beard is a free software; you can redistribute it and/or modify it
+# under the terms of the Revised BSD License; see LICENSE file for
+# more details.
+
+"""Blocking for clustering estimators.
+
+.. codeauthor:: Gilles Louppe <g.louppe@cern.ch>
+
+"""
+import numpy as np
+
+from sklearn.base import BaseEstimator
+from sklearn.base import ClusterMixin
+from sklearn.base import clone
+from sklearn.utils import column_or_1d
+
+
+def _single(X):
+    return np.ones(len(X), dtype=np.int)
+
+
+class BlockClustering(BaseEstimator, ClusterMixin):
+
+    """Implements blocking for clustering estimators.
+
+    Meta-estimator for grouping samples into blocks, within each of which
+    a clustering base estimator is fit. This allows to reduce the cost of
+    pairwise distance computation from O(N^2) to O(sum_b N_b^2), where
+    N_b <= N is the number of samples in block b.
+
+    Attributes
+    ----------
+    labels_ : ndarray, shape (n_samples,)
+        Array of labels assigned to the input data.
+        if partial_fit is used instead of fit, they are assigned to the
+        last batch of data.
+    """
+
+    def __init__(self, blocking="single", base_estimator=None):
+        """Initialize.
+
+        Parameters
```

Comment:
Yes indeed, but you may also simply access the docstring from an IPython shell or notebook, where no such formatting is done. In this case, having such structure is really helpful.